relay-server/src/actors/project_cache.rs

```diff
@@ -550,26 +550,45 @@ impl ProjectCacheBroker {
     }
 
     fn handle_get(&mut self, message: GetProjectState, sender: ProjectSender) {
-        self.get_or_create_project(message.project_key)
+        let request_update = self
+            .get_or_create_project(message.project_key)
             .get_state(sender, message.no_cache);
+
+        if let Some(request_update) = request_update {
+            self.handle_request_update(request_update)
+        }
```

Comment:
To be honest, I am not sure whether this is actually better than passing the `Addr`s to `Project`. I feel that it will be easy to forget to send the update request in future refactors. I am in favor of reverting back to https://github.com/getsentry/relay/pull/1928/commits/3940b2ce83cf1fed2d54d8c8ce542944ba0ed029 and merging that version.

Alternatively, we could pass a callback or an `Addr` into `Project::get_state` such that the project takes care of sending the `RequestUpdate` again (similar as you do with `flush_buckets`).

Reply:
Right now the service does not know address of themselves, which can be fixed of course.
Passing address as a parameter might be better here, that they using this function will force the caller to provide a correct type.

Reply:
changed this in [81baec3](https://github.com/getsentry/relay/pull/1928/commits/81baec31435fd9712919fb15ec480cb8c4ad1530)

Reply:
Thanks, it looks much safer now!